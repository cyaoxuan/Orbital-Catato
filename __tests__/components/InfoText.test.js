import {
    IconTextField,
    IconTextFieldRow,
    KeyTextField,
} from "../../app/components/InfoText";
import { cleanup, render } from "@testing-library/react-native";

afterEach(cleanup);

describe("<IconTextField />", () => {
    it("renders successfully", () => {
        const iconTextField = render(<IconTextField />);
        expect(iconTextField).toBeDefined();
    });

    it("renders correctly", () => {
        const tree = render(<IconTextField />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders icon", () => {
        const { getByTestId } = render(<IconTextField />);
        const icon = getByTestId("icon");
        expect(icon).toBeDefined();
    });

    it("displays field and info prop", () => {
        const { queryByText } = render(
            <IconTextField field="Hello" info="World!" />
        );
        expect(queryByText("Hello")).not.toBeNull();
        expect(queryByText("World!")).not.toBeNull();
    });
});

describe("<IconTextFieldRow />", () => {
    it("renders successfully", () => {
        const iconTextField = render(<IconTextFieldRow />);
        expect(iconTextField).toBeDefined();
    });

    it("renders correctly", () => {
        const tree = render(<IconTextFieldRow />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders icon", () => {
        const { getByTestId } = render(<IconTextFieldRow />);
        const icon = getByTestId("icon");
        expect(icon).toBeDefined();
    });

    it("displays field and info prop", () => {
        const { queryByText } = render(
            <IconTextFieldRow field="Hello: " info="World!" />
        );
        expect(queryByText("Hello: World!")).not.toBeNull();
    });
});

describe("<KeyTextField />", () => {
    it("renders successfully", () => {
        const keyTextField = render(<KeyTextField />);
        expect(keyTextField).toBeDefined();
    });

    it("renders correctly", () => {
        const tree = render(<KeyTextField />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("displays field prop", () => {
        const { queryByText } = render(<KeyTextField field="Hello" />);
        expect(queryByText("Hello")).not.toBeNull();
    });

    it("displays info prop", () => {
        const { queryByText } = render(<KeyTextField info="World" />);
        expect(queryByText("World")).not.toBeNull();
    });
});
