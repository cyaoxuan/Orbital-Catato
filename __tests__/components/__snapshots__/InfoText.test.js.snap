// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconTextField /> renders correctly 1`] = `
<View>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 24,
          },
          {
            "marginHorizontal": 4,
          },
          {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
      testID="icon"
    >
      
    </Text>
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {
            "writingDirection": "ltr",
          },
          {
            "flex": 1,
            "fontFamily": "Nunito-Bold",
          },
        ]
      }
    >
      Field: 
    </Text>
  </View>
  <View
    style={
      {
        "paddingLeft": 24,
      }
    }
  >
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {
            "writingDirection": "ltr",
          },
          {
            "flex": 1,
            "fontFamily": "Nunito-Medium",
          },
        ]
      }
    >
      Info
    </Text>
  </View>
</View>
`;

exports[`<IconTextFieldRow /> renders correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 24,
        },
        {
          "marginHorizontal": 4,
        },
        {
          "fontFamily": "ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
    testID="icon"
  >
    
  </Text>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        {
          "flex": 1,
          "fontFamily": "Nunito-Bold",
        },
      ]
    }
  >
    Field: 
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {
            "writingDirection": "ltr",
          },
          {
            "flex": 1,
            "fontFamily": "Nunito-Medium",
          },
        ]
      }
    >
      Info
    </Text>
  </Text>
</View>
`;

exports[`<KeyTextField /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F8F9FD",
      "borderRadius": 10,
      "height": 80,
      "padding": 8,
      "width": 85,
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "400",
          "letterSpacing": 0.25,
          "lineHeight": 20,
        },
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        {
          "flex": 1,
          "fontFamily": "Nunito-Medium",
        },
      ]
    }
  >
    Field
  </Text>
  <Text
    style={
      [
        {
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "400",
          "letterSpacing": 0.15,
          "lineHeight": 24,
        },
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        {
          "flex": 1,
          "fontFamily": "Nunito-Bold",
        },
      ]
    }
  >
    Info
  </Text>
</View>
`;
